src/lib.rs

```diff
@@ -40,6 +41,55 @@ pub enum Difference {
     Rem(String),
 }
 
+/// The information about a full changeset
+pub struct Changeset {
+    /// An ordered vector of `Difference` objects, coresponding
+    /// to the differences within the text
+    pub diffs: Vec<Difference>,
+    /// The split used when creating the `Changeset`
+    /// Common splits are `""` for char-level, `" "` for word-level and `"\n"` for line-level.
+    pub split: String,
+    /// The edit distance of the `Changeset`
+    pub distance: i32,
+}
+
+impl Changeset {
+    /// Calculates the edit distance and the changeset for two given strings.
+    /// The first string is assumed to be the "original", the second to be an
+    /// edited version of the first. The third parameter specifies how to split
+    /// the input strings, leading to a more or less exact comparison.
+    ///
+    /// Common splits are `""` for char-level, `" "` for word-level and `"\n"` for line-level.
+    ///
+    /// Outputs the edit distance (how much the two strings differ) and a "changeset", that is
+    /// a `Vec` containing `Difference`s.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use difference::Changeset;
+    /// use difference::Difference;
+    ///
+    /// let changeset = Changeset::new("test", "tent", "");
+    ///
+    /// assert_eq!(changeset.diffs, vec![
+    ///     Difference::Same("te".to_string()),
+    ///     Difference::Rem("s".to_string()),
+    ///     Difference::Add("n".to_string()),
+    ///     Difference::Same("t".to_string())
+    /// ]);
+    /// ```
+    pub fn new(orig: &str, edit: &str, split: &str) -> Changeset {
+        let (dist, common) = lcs(orig, edit, split);
+        Changeset {
+            diffs: merge(orig, edit, &common, split),
+            split: split.to_string(),
```

Comment:
it's necessary to implement `Display` on the `Changeset` object -- without it the display method wouldn't know whether to new newlines, spaces, etc.